### PR TITLE
Add save_excel and save_csv methods to FitData object 

### PR DESCRIPTION
**Pull request by OhadNir9**
Solves #52 and #53

### CODE REVIEW

File: src/eddington/fit_data.py [hunk 1]
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
             return True
         except ValueError:
             return False
+
+    def __make_sheet(self, sheet=None):
```
Comment (saroad2) [inline src/eddington/fit_data.py:440]: There is no separate logic here that may be called in more than one place.

You can move the entire section to the `save_excel` method

File: src/eddington/fit_data.py [hunk 2]
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
...
+        for row in zip(*columns):
+            worksheet.append(row)
+
+        return workbook, worksheet
+
+    def save_excel(self, output_directory, name=None, sheet=None):
+        """
+        Save :class:`FitData` to xlsx file.
+
+        :param output_directory: str or Path.
```
Comment (saroad2) [inline src/eddington/fit_data.py:461]: Specify the type of the parameter in a `:type:` section.

File: src/eddington/fit_data.py [hunk 3]
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
...
+
+    def save_csv(self, output_directory, name=None):
+        """
+        Save :class:`FitData` to csv file.
+
+        :param output_directory: str or Path.
+         Path to the directory for the new excel file to be saved.
+        :param name: Optional. The name of the file, without the .csv suffix
+        """
+        worksheet = self.__make_sheet()[1]
```
Comment (saroad2) [inline src/eddington/fit_data.py:483]: There is no need to make a worksheet here, especially if you drop it emidiatly after.

I would recommend you use [DictWriter](https://docs.python.org/3/library/csv.html#csv.DictWriter) in order to write the csv file.

File: src/eddington/fit_result.py [hunk 4]
```diff
@@ -9,7 +9,7 @@
 
 
 @dataclass(repr=False)  # pylint: disable=too-many-instance-attributes
-class FitResult:
+class FitResult:  # pylint: disable=too-many-instance-attributes
```
Comment (saroad2) [inline src/eddington/fit_result.py:12]: This fix is not relevant to this new feature and it should be in a different PR

File: tests/fit_data/test_save_data_to_file.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
+from pathlib import Path
```
Comment (saroad2) [inline tests/fit_data/test_save_data_to_file.py:1]: Those are actually very solid **integration tests**.

One key element of **unit tests** is that they don't interact with the operating system (aka, read and write files).
*Integration tests* however, do exactly that. They test the software end-to-end and see that the different components combine nicely together.

Please keep those tests as they are, but move them to the *integration_tests* directory (if it's missing in your branch, update the master branch from upstream, and merge it into your feature branch. I added this directory yesterday).

After you do that, write new unit tests, but use [mocks](https://realpython.com/python-mock-library/) in order to simulate writing to a file.

Mocks can sometimes be perceived as a little bit hard concept for developers who are not familiar with them. Please feel free to say that you rather I will write the unit tests for you, and I'll step right in.

**Review by saroad2 — CHANGES_REQUESTED** [review]
Thank you for your contribution.

I left some comments. Please feel free to return a feedback or ask for help if needed.

File: src/eddington/fit_data.py [hunk 6]
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
...
+        workbook = self.__make_sheet(sheet)[0]
+
+        if name:
+            path = Path(output_directory / f"{name}.xlsx")
+        else:
+            path = Path(output_directory / "FitData.xlsx")
+
+        workbook.save(path)
+
+    def save_csv(self, output_directory, name=None):
```
Comment (saroad2) [inline src/eddington/fit_data.py:475]: use default `name="fit_data.csv"`

File: src/eddington/fit_data.py [hunk 7]
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
...
+         Path to the directory for the new excel file to be saved.
+        :param name: Optional. The name of the file, without the .csv suffix
+        """
+        worksheet = self.__make_sheet()[1]
+        data = worksheet.rows
+
+        if name:
+            path = Path(output_directory / f"{name}.csv")
+        else:
+            path = Path(output_directory / "FitData.csv")
```
Comment (saroad2) [inline src/eddington/fit_data.py:489]: Remove this if-else statement and use default value in the method signature

File: src/eddington/fit_data.py [hunk 8]
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
...
+        columns = list(self.data.values())
+
+        worksheet.append(headers)
+
+        for row in zip(*columns):
+            worksheet.append(row)
+
+        return workbook, worksheet
+
+    def save_excel(self, output_directory, name=None, sheet=None):
```
Comment (saroad2) [inline src/eddington/fit_data.py:457]: use `name="fit_data.xlsx"`

File: src/eddington/fit_data.py [hunk 9]
```diff
@@ -436,3 +436,67 @@ def __is_number(cls, string):
...
+         Path to the directory for the new excel file to be saved.
+        :param name: Optional. The name of the file, without the .xlsx suffix
+        :param sheet: Optional. Name of the sheet that the data will be saved to.
+        """
+        workbook = self.__make_sheet(sheet)[0]
+
+        if name:
+            path = Path(output_directory / f"{name}.xlsx")
+        else:
+            path = Path(output_directory / "FitData.xlsx")
```
Comment (saroad2) [inline src/eddington/fit_data.py:471]: Remove if-else statement and use method signature default instead.

**Review by saroad2 — CHANGES_REQUESTED** [review]

**Comment by OhadNir9** [comment]
I'm having trouble with the mock concept, if it's ok I prefer you'll do it. I moved the test file to integration_tests.
Moreover, as you requested I removed the pylint comment from fit_result.

**Review by saroad2 — APPROVED** [review]
Thank you for your contribution!

I added the missing unit tests and fixed some small issues.

**Pull request merged**